### ISSUE TITLE
Unable to overwrite request header with application/json content

### ISSUE THREAD

**Issue by cguy** [apideck-libraries/portman#458]
# Issue

Portman doesn't seem to be able to inject application/json content in header

## How to test

You have to download the stackblitz https://stackblitz.com/edit/node-rq2nbx?file=README.md repo because Portman doesn't seem to work on stackblitz.

Then, open **two** terminals.

In the first :

`npm ci && npm start`

Then, in the second :

`npx portman --cliOptionsFile ./options.json --runNewman --newmanRunOptions '{"insecure": true, "abortOnFailure": false}' -b http://localhost:3000`

## Expected behavior

View in express logs the header `x-my-json` the value `"{\"type\": \"foo\", \"color\": \"white\"}"`

```json
{
    "x-my-header": "test2",
    "x-my-json": "{\"type\": \"foo\", \"color\": \"white\"}",
    "user-agent": "PostmanRuntime/7.29.0",
    "accept": "*/*",
    "cache-control": "no-cache",
    "postman-token": "8c7c6e23-990e-4b08-bf4f-cfaa1d5ec530",
    "host": "localhost:3000",
    "accept-encoding": "gzip, deflate, br",
    "connection": "keep-alive"
}
```

## Current behavior

The header `x-my-json` doesn't have any value.

```json
{
    "x-my-header": "test2",
    "x-my-json": "",
    "user-agent": "PostmanRuntime/7.29.0",
    "accept": "*/*",
    "cache-control": "no-cache",
    "postman-token": "8c7c6e23-990e-4b08-bf4f-cfaa1d5ec530",
    "host": "localhost:3000",
    "accept-encoding": "gzip, deflate, br",
    "connection": "keep-alive"
}
```

## Part of explanation

My guess is that Portman is not able to fulfill an example value when we describe a json content instead of a string schema.

Then, in the portman source code (https://github.com/apideck-libraries/portman/blob/main/src/application/overwrites/overwriteRequestHeaders.ts#L35), we overwrite request header value only if we already have a value in the header.

What should I do ? (using object schema instead of application/json content is not an option)

**Comment by thim81** [comment]
hi @cguy 
That is a good catch.
For the other overwrite methods, we do support the "insert" option.
It seems we didn't align this across the headers.

I'll tag this as a bug and try to pick it up, once we find some time.

**Comment by cguy** [comment]
Hi @thim81, 

Actually, I think I saw the insert flag for request headers. But since my header is defined in my open api spec, it won't be inserted. (maybe you're talking about something else ?).

Is there something I can do?

**Comment by thim81** [comment]
There is a Portman config property for `overwriteRequestHeaders`: 
> insert (Boolean true/false | Default: true) : Insert the additional request header in Postman that are not present in OpenAPI.

Which does not seem to do what it is supposed to do, hence that is why it seems like a bug.
In https://github.com/apideck-libraries/portman/blob/76b44230e836d12102855e03fe25c0d46d1ddbf9/src/application/overwrites/overwriteRequestHeaders.ts#L70 Portman should insert any headers

**Comment by cguy** [comment]
Oh, in your opinion this is an insert issue, not an overwrite issue?

Maybe I can start a PR and we iterate about the technical solution? What do you think?

**Comment by thim81** [comment]
I'm not 100% sure, but looking at your config, I think the reason will mostly be related to the header having a JSON value

```yaml
- in: header
          name: x-my-json
          required: true
          content:
            application/json:
              schema:
                example: "{\"type\": \"foo\", \"color\": \"white\"}"
                type: object
                required:
                  - "color"
                properties:
                  type:
                    type: string
                  color:
                    type: string
                    pattern: "(black|white)"
```

**Comment by cguy** [comment]
Yeah, but I am not sure to understand why it can be an issue. 

The open API specification allow this use case : https://swagger.io/specification/v3/#parameter-object 

> For more complex scenarios, the [content](https://swagger.io/specification/v3/#parameter-content) property can define the media type and schema of the parameter. A parameter MUST contain either a schema property, or a content property, but not both. When example or examples are provided in conjunction with the schema object, the example MUST follow the prescribed serialization strategy for the parameter.

But, where I agree with you is that the open api parser may not be able to parse content attribute in JSON, leading the issue with a header with empty string value "". This behavior explains why portman is not able to overwrite header value.

So, in my opinion there are some options (non exhaustive, could be cumulative):
- investigate the open api parser and probably fix it to let it fulfill the header value
- fix portman to let the possibility to overwrite a header that contains an empty string value (the if statement in https://github.com/apideck-libraries/portman/blob/main/src/application/overwrites/overwriteRequestHeaders.ts#L35 could be as followed:

Before
```typescript
if (overwriteItem?.value !== undefined && pmHeader?.value) {
      const orgValue = pmHeader.value
      let newValue = overwriteItem.value

      if (overwriteItem.overwrite === false) {
        newValue = orgValue + newValue
      }
      pmHeader.value = newValue || 'boolean' === typeof newValue ? `${newValue}`.toString() : ''
    }
``` 

After
```typescript
if (overwriteItem?.value !== undefined) {
      const orgValue = pmHeader.value
      let newValue = overwriteItem.value

      if (overwriteItem.overwrite === false && orgValue) {
        newValue = orgValue + newValue
      }
      pmHeader.value = newValue || 'boolean' === typeof newValue ? `${newValue}`.toString() : ''
    }
```

**Comment by cguy** [comment]
Hello @thim81, do you know when the next version of portman will be released (in a few days, in a few weeks, in a few months...)?

**Comment by thim81** [comment]
hi @cguy 
I have it on my agenda for this, to bump some dependencies and create a new release with all the changes.

**Comment by thim81** [comment]
hi @cguy 

We just released Portman 1.21.0, which includes your fix.
Thanks again for taking the time to report the issue with all the details and creating the PR that fixed it.

**Comment by cguy** [comment]
Thank you for your reactivity!
